### PR TITLE
[Draft] move programs to 1 repo

### PR DESCRIPTION
**Pull request by anhphi257**
- Move program from https://github.com/step-finance/reward-pool to dual-farming
- Move program from https://github.com/mercurial-finance/mer-staking to staking
- Change current program to single-farming
- Update anchor version to 0.24.2
- Update libs following new version

### CODE REVIEW

**Comment by anhphi257** [comment]
Not done: 
- /// CHECK: more detail explaination
- Make unittests run

File: programs/dual-farming/src/calculator/mod.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+use crate::*;
```
Comment (andrewsource147) [inline programs/dual-farming/src/calculator/mod.rs:1]: Could you check whether there are new updates from step-finance reward pools?

File: programs/dual-farming/src/calculator/pool_v1.rs [hunk 2]
```diff
@@ -0,0 +1,140 @@
+use crate::calculator::*;
```
Comment (andrewsource147) [inline programs/dual-farming/src/calculator/pool_v1.rs:1]: Pls remove pool_v1, and rename pool_v2
Step finance has to update program, so they create v1, v2, but we deploy new program, then no need to use different versions

Comment (anhphi257) [reply programs/dual-farming/src/calculator/pool_v1.rs]: Updated

File: programs/dual-farming/src/calculator/mod.rs (hunk 1 above)
Comment (anhphi257) [reply programs/dual-farming/src/calculator/mod.rs:1]: removed logging following step-finance reward-pool updates

**Pull request closed without merging**